### PR TITLE
make DismissItem call ConsumeMessage properly

### PR DESCRIPTION
**Pull request by oconnor663**
Previously we were calling the wrong end of the API. BroadcastMessage is
what is supposed to get called when the *server* informs us of a
message.

r? @maxtaco @mmaxim 

This is not yet working. The client RPC call seems to hang forever, and I can't see anything at all in the gregord logs. Probably something thinks it's not connected yet. I will debug with you guys tomorrow.

### CODE REVIEW

**Comment by maxtaco** [comment]
Well shoot.  This looks like it would do the trick.  I would turn on RPC debugging.

File: go/service/gregor.go [hunk 1]
```diff
@@ -645,6 +645,8 @@ func (g *gregorHandler) connectNoTLS(uri *rpc.FMPURI) error {
 }
 
 func (g *gregorHandler) DismissItem(id gregor.MsgID) error {
+	g.G().Log.Debug("gregorHandler.DismissItem called with MsgID %s", id.String())
```
Comment (maxtaco) [inline go/service/gregor.go]: do you need to grab a lock here?

**Comment by mmaxim** [comment]
Yeah this looks like a bug with `g.cli`. I'm not sure why though, we can probably figure out this quickly tomorrow morning.

**Comment by maxtaco** [comment]
@oconnor663 did you try adding debugging into the connection.go library?

**Comment by maxtaco** [comment]
We also need a test for this, since clearly the tests weren't good enough to catch this bug.

**Comment by mmaxim** [comment]
Yeah I realize this would be the first use of `g.cli` in gregor.go, all other calls into gregord use the client from the RPC library in `OnConnect`. This is basically the same code gregord uses for authd, so I'm surprised there is a problem. Agreed about the test though.

**Comment by mmaxim** [comment]
Actually how are you testing this @oconnor663, is it getting stuck after syncing from gregord and replaying a message?

**Comment by mmaxim** [comment]
The point is if this function gets called from a code path originating in `OnConnect`, it won't work.

**Comment by maxtaco** [comment]
@mmaxim seems like the intention is to call this when the UI closes the window.

**Comment by maxtaco** [comment]
See go/engine/track_token

**Comment by oconnor663** [comment]
Ok Mike helped me figure this out. I think the problem was that my testing was running in OnConnect, when the g.cli object was in a bad state. I've filed https://keybase.atlassian.net/browse/CORE-3136 to think about that more.

Once I tested it right, I found a double encoding bug. That's fixed now.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: No visual changes found as a result of these commits.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: No visual changes found as a result of these commits.

**Comment by oconnor663** [comment]
One more problem I'm seeing in the logs: we're broadcasting the dismissal locally twice. Everything handles that ok, but I'm not sure why it happens.

**Comment by oconnor663** [comment]
@mmaxim and I decided that the double dismiss was a gregord problem where it's accidentally rebroadcasting messages to itself. This PR should be good to go.

**Comment by mmaxim** [comment]
👍

**Comment by oconnor663** [comment]
landed with 7cfb44a382a4be94d0326d54761dcfaad018510f

**Pull request closed without merging**